### PR TITLE
Fix a problem with Import Transition List / Assay Library, wherein user isn't given the chance to deal with errors after click on OK if errors were previously reviewed…

### PR DESCRIPTION
**Pull request by bspratt**
… (unintended side effect of AssociateProteins work trying to avoid redundant work).

Updated AbstractFunctionalTest.ImportAssayLibraryOrTransitionList() to check this scenario handling transition lists with errors:
                 In transition list import column select dialog, hit OK
                 Get the error dialog, hit cancel - takes you back to column select dialog as it should
                 In column select, hit "Check for Errors"
                 Get the error dialog, hit OK - takes you back to column select as it should
                 In column select dialog, hit OK - skips right over the expected error check dialog, which is bad

### CODE REVIEW

**Comment by bspratt** [comment]
The issue centers around the "IsCheckForErrorsNecessary(isAssociateProteins)" call - I guess there's more than one set of code for associate proteins.

**Comment by bspratt** [comment]
Yeah, oops, not your work. I'll get a different reviewer.

**Review by nickshulman — APPROVED** [review]
I would recommend making one small change but otherwise looks good to me.

File: pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs [hunk 1]
```diff
@@ -1276,71 +1276,82 @@ private void InitializeRadioButtons()
...
-                    MissingEssentialColumns = new List<string>();
-                    if (radioPeptide.Checked)
+                    using var longWaitDlg = new LongWaitDlg
+                        {Text = Resources.ImportTransitionListColumnSelectDlg_CheckForErrors_Checking_for_errors___};
+                    longWaitDlg.PerformWork(this, 1000, progressMonitor =>
+                    {
+
+                        var columns = Importer.RowReader.Indices;
+                        MissingEssentialColumns = new List<string>();
+                        if (radioPeptide.Checked)
```
[the remark this answers is not in the document]
Comment (bspratt) [reply pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs:1311]: Nice catch, thanks!

[the remark this answers is not in the document]
Comment (bspratt) [reply pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs:1311]: There are actually a handful of control accesses in there, I'll store those as well.

**Pull request merged**